File: python/arcade-intro/06_Rains of Reason/027_variableName.py
import re

def variableName(name):
    '''
    Correct variable names consist only of Latin letters, digits and underscores 
    and they can't start with a digit.
    Check if the given string is a correct variable name.
    '''
    p = re.compile("^[a-zA-Z_$][a-zA-Z_$0-9]*$")
    return p.match(name) is not None

print(variableName("asdf"))
print(variableName("0sdf"))
print(variableName("____f24fsdf__"))
print(variableName("sgwda_asdfasdf__asdf__"))

File: arcade/go/arcade-intro/06_Rains of Reason/027_variableName.go
func variableName(name string) bool {
	if len(name) <= 0 {
		return false
	}

	// check if first char is a number
	if name[0] >= 48 && name[0] <= 57 {
		return false
	}

	for _, s := range name {
		if s != 95 && !(s >= 65 && s <= 90) && !(s >= 97 && s <= 122) && !(s >= 48 && s <= 57) {
			return false
		}
	}
	return true
}
